test/test-network/systemd-networkd-tests.py

```diff
@@ -6839,6 +6839,51 @@ def check(self, ipv4, ipv6):
         check(self, True, False)
         check(self, False, True)
         check(self, False, False)
+    
+    def test_dhcp_client_default_use_domains(self):        
+        def check(self, ipv4, ipv6):
+            os.makedirs(networkd_conf_dropin_dir, exist_ok=True)
+            with open(os.path.join(networkd_conf_dropin_dir, 'default_use_domains.conf'), mode='w', encoding='utf-8') as f:
+                f.write('[DHCPv4]\nUseDomains=')
+                f.write('yes' if ipv4 else 'no')
+                f.write('\n[DHCPv6]\nUseDomains=')
+                f.write('yes\n' if ipv6 else 'no\n')
```

Comment:
Not a big matter, but this is slightly asymmetric. Please move `\n` before `[DHCPv6]` to the first `yes` or `no`.
```python
                f.write('[DHCPv4]\nUseDomains=')
                f.write('yes\n' if ipv4 else 'no\n')
                f.write('[DHCPv6]\nUseDomains=')
                f.write('yes\n' if ipv6 else 'no\n')```